src/validation.cpp

```diff
@@ -2637,7 +2642,16 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
             }
 
             const auto applyCustomTxTime = GetTimeMicros();
-            const auto res = ApplyCustomTx(accountsView, view, tx, chainparams.GetConsensus(), pindex->nHeight, pindex->GetBlockTime(), nullptr, i, evmContext);
+            uint64_t gasUsed{};
+            const auto res = ApplyCustomTx(accountsView, view, tx, chainparams.GetConsensus(), pindex->nHeight, gasUsed, pindex->GetBlockTime(), nullptr, i, evmContext);
+
+            totalGas += gasUsed;
+            if (totalGas > MAX_BLOCK_GAS_LIMIT) {
+                return state.Invalid(ValidationInvalidReason::CONSENSUS,
+                                     error("%s: ApplyCustomTx failed. Gas limit: %d Gas used: %d", __func__, MAX_BLOCK_GAS_LIMIT, totalGas),
+                                     REJECT_CUSTOMTX, "over-gas-limit");
+            }
+
```

Comment:
If the TX goes above the MAX_BLOCK_GAS_LIMIT, we need to reject it and try the following ones.
The if block needs to subtract this TX used_gas so the following ones have a chance to get in the block

Reply:
How about this?

- See if there's atleast one EVM TX that's included. If it is, keep the same logic, skip the rest and fail the block. 
- If it's not, reject it 

This prevents head of line blocking by a single TX, but safely rejects others, without us having to worry about reordering for now.